doc-source/quick-start.rst

```diff
@@ -64,6 +64,55 @@ As an example, the format for ``controls.json`` is as follows::
     "chk_pkg.chk_cat_bar.checks.chk_module_bar.BarCheckClass": ["accred.one", "accred.two"]
   }
 
+Agents
```

Comment:
Yes, all valid points.
For the first two items, not providing the agent configuration (`agent_name` & `agent_signing_key`) would result in the current behaviour (i.e. no signing / no storing evidence under `agents/...`). This would be appropriate for the "CI agent".
For the third, I'm happy to reconsider the evidence format. Storing the digest/signature in the index file (or a separate meta file) sounds good. From my perspective, I wanted a single file I could present to an external auditor, something that equates to a gpg clearsign (but using RSA). This is perhaps a selfish requirement; I'm happy to go with the majority.

Reply:
> For the first two items, not providing the agent configuration (agent_name & agent_signing_key) would result in the current behaviour (i.e. no signing / no storing evidence under agents/...). This would be appropriate for the "CI agent".

However, if we decide to enable signing of evidence from the CI job, which is what I recommend we do in the future, then we'd need a way to tell it to use the evidence in the root folder as opposed to the agent folder. Hence `use_agent_dir`.

For `agent_enable_signing`, I agree that this attribute is not strictly needed, as the only reason to configure an agent is to enable signing. It just seemed more consistent with thinking of these things (agents and signed evidence) as separate features.

> For the third, I'm happy to reconsider the evidence format. Storing the digest/signature in the index file (or a separate meta file) sounds good. From my perspective, I wanted a single file I could present to an external auditor, something that equates to a gpg clearsign (but using RSA). This is perhaps a selfish requirement; I'm happy to go with the majority.

Having a single file to present to auditors definitely keeps things simple, and makes it very clear what is being signed. I just think that we'll run into problems with it quickly once we start to expand what gets signed. Better to use a more generic approach.

Reply:
I understand your point about `use_agent_dir` now. I'll address this. I do like the digest/signature in the `index.json` file idea. I'll try and get that working.